readme.md

```diff
@@ -1,15 +1,95 @@
+
+The current updating solution relies on an Amazon Lambda function to hit the corresponding API endpoints to trigger the update. Internal timed tasks would not work in a monolith architecture when scaling occurs, as the action would be duplicated .
+
+The user account system relies on _Amazon Web Services'_ ([AWS](https://aws.amazon.com/)) _Identity & Access Management_ ([IAM](https://aws.amazon.com/iam/)) system by using [_Amazon Cognito_](https://aws.amazon.com/cognito/) through the provided [_Amazon Amplify_](https://aws.amazon.com/amplify/) javascript library. As an aside, the client application uses assets that are hosted using [AWS S3 Storage](https://aws.amazon.com/s3/). 
+
+The whole system is hosted using [_AWS Elastic Beanstalk_](https://aws.amazon.com/elasticbeanstalk/) and [_Amazon EC2_](https://aws.amazon.com/ec2/).
+
+The system also uses a custom Amazon Lambda function with a post-confirmation trigger for copying user-data (AWS-ID and username) to the system database. The trigger happens when a new user confirms their email through the provided sign-up system, simply sending a POST request with the data to the ```/users``` endpoint. 
+
+For an efficient development process, Continuous Deployment was implemented through [_AWS CodePipeline_](https://aws.amazon.com/codepipeline/).
+
+Furthermore, AWS hosts the [MongoDB](https://www.mongodb.com/) database for the server application. Due to technical difficulties the connection string to the  MongoDB instance is stored in plain text in the source code. However, the MongoDB instance only has the production server address whitelisted.  
+
+For emitting push notification, the system leverages a [development kit for Expo](https://github.com/expo/expo-server-sdk-node) along with Expo's notification infrastructure, owing to the fact that the client application is written in React Native, which also uses [Expo](https://docs.expo.io/).
+
+The server application includes several pieces of vestigial or unused functionality, and functionality that should be authorized only for a system adminstrator, like adding new achievments etc.
+
+The repository includes an unmerged branch containing validation logic for request bodies for client facing routes.
+
+## Features
+
+<img src="https://user-images.githubusercontent.com/47078189/99588697-482e9d80-29eb-11eb-806b-a7f48eaa8635.png" width="500">
+
+- Served auto-generated Swagger API documentation
```

Comment:
Should we add a link to the docs here? I.e.: http://ec2-18-192-82-31.eu-central-1.compute.amazonaws.com/docs/